app/lib/account/backend.dart

```diff
@@ -199,26 +200,62 @@ class AccountBackend {
     return likes.map((Like l) => l.package).toList();
   }
 
-  /// Creates an a package like entry for the given [user] and [package].
+  /// Creates and returns a package like entry for the given [user] and
+  /// [package], and increments the 'likes' property on [package].
   Future<Like> likePackage(User user, String package) async {
-    final newLike = Like()
-      ..parentKey = user.key
-      ..id = package
-      ..created = DateTime.now().toUtc();
-    await _db.commit(inserts: [newLike]);
-    return newLike;
+    final like = await _db.withTransaction<Like>((tx) async {
+      final packageKey = _db.emptyKey.append(Package, id: package);
+      final p = await tx.lookupValue<Package>(packageKey);
```

Comment:
```suggestion
      final p = await tx.lookupValue<Package>(packageKey, orElse: () => null);
```

Hmm, I think we should add test cases for "package does not exist" and things like that.